src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -179,17 +177,17 @@ public Claim saveClaim(
 
         Claim issuedClaim;
         try {
-            Long prePaymentClaimId = caseRepository.getOnHoldIdByExternalId(externalId, authorisation);
-
-            LocalDateTime now = LocalDateTimeFactory.nowInLocalZone();
+            caseRepository.getClaimByExternalId(externalId, authorisation).ifPresent(claim -> {
+                throw new ConflictException(
```

Comment:
We are throwing this exception here and catching straight away in the same method, and then calling `caseRepository.getClaimByExternalId(externalId, authorisation)` in catch block few lines down. This is not right.

Reply:
Will there be any change in data for the claim between line:201 and line:226? Both are returning the same claim information.

The second call in catch block to get the claim will return exactly the same claim object.

Reply:
Throwing conflictexception is to explicitly make sure by coding that its unfavourable situation. Handling how you want should be in catch block where we fetch the claim again and handle the exception. The catch block should not get information or use the information from what it has got from Try block.  But I agreed to change it as what @sabahirfan , @dharmendrak  and @andrewwa-kainos has agreed. 

There will be a separate PR to do the cleanup.

Reply:
The additional unnecessary get call will decrease the performance while we are on CCD.

Reply:
The point is even the conflictexception should not occur. These check should be even in controller and throw exception rather than coming till repository. This is leak in transaction. In CCD this wont even happen.